content/chainguard/migration-guides/tutorial/index.md

```diff
@@ -0,0 +1,732 @@
+been changed to use [gunicorn](https://gunicorn.org/) as [uwsgi has entered "maintenance
+mode"](https://github.com/unbit/uwsgi).
+
+Build the final image:
+
+```bash
+docker build -t identidock-cg .
+```
+
+And take a look at it:
+
+```bash
+docker images identidock-cg
+REPOSITORY      TAG       IMAGE ID       CREATED          SIZE
+identidock-cg   latest    30a424aa8a51   15 seconds ago   93.6MB
+```
+
+Run a scan with grype:
+
+
+```bash
+grype docker:identidock-cg
+✔ Vulnerability DB                [updated]
+✔ Loaded image                                                                                                                                                                    identidock-cg:latest
+✔ Parsed image                                                                                                                 sha256:30a424aa8a51548491ea3c152383ce2c16a801859e9e8e252b147162ac0b637e
+✔ Cataloged contents                                                                                                                  40024c9c378c73d52c9e62f32fcf6d0d328d8d14a66a328df8d6fc0278a4a2e8
+├── ✔ Packages                        [46 packages]
+├── ✔ File digests                    [2,020 files]
+├── ✔ File metadata                   [2,020 locations]
+└── ✔ Executables                     [141 executables]
+✔ Scanned for vulnerabilities     [0 vulnerability matches]
+├── by severity: 0 critical, 0 high, 0 medium, 0 low, 0 negligible
+└── by status:   0 fixed, 0 not-fixed, 0 ignored
+```
+
+The result of all these changes is that the production image is only 93.6 MB (down from 1.05GB, so
+an enormous saving of just under a GB) and has 0 CVEs (down from 980). This is a huge improvement!
+
+Further information on using Chainguard Images with Python can be found in our
+[guide](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/python/).
+
+
+## Updating Redis
+
+This migration is the most straightforward. We're not making any changes to the application image,
+so all we need to do is directly update the reference to `redis:7` in the Docker Compose file to
+`cgr.dev/chainguard/redis`. The new image requires no extra configuration but we go from a 139 MB
+image with 137 vulnerabilities to a 23MB image with 0 CVEs (again according to Grype).
+
+Full details on the new Docker Compose file are found in the next section.
+
+## Updating Docker Compose
+
+In the top level directory of the repo, replace the `docker-compose.yaml` with the following:
+
+
+```yaml
+name: identidock
+
+services:
+  frontend:
+    build: identidock
+    ports:
+      - "9090:9090"
+
+  dnmonster:
+    build: dnmonster
+
+  redis:
+    image: cgr.dev/chainguard/redis
+```
+
+If you now run `docker compose up –build`, you should have a working application that can be reached on port 9090:
+
+![Screenshot of running application](app.png "Screenshot of running application")
```

Comment:
This final piece works for me despite the weird build piece for the dnmonster part.

Reply:
Good. That was just a typo from me.